scripts/sbom_scraper.sh

```diff
@@ -133,19 +159,179 @@ SECRET=$(cat "${CLIENTSECRET_FILE}")
     OUTPUT="${TEMPDIR}/${OUTFILE}"
     ${SYFT} -q packages -o "${FORMAT}" "${DOCKER_IMAGE}"> "${OUTPUT}"
 else
     OUTPUT="${DOCKER_IMAGE}"
 fi
 
+# ----------------------------------------------------------------------------
+# Update SBOM including NTIA minimum elments
+# ----------------------------------------------------------------------------
+ORIG_COMPONENT_NAME=$(xq -r .bom.metadata.component.name "$OUTPUT")
+ORIG_COMPONENT_VERSION=$(xq -r .bom.metadata.component.version "$OUTPUT")
+COMPONENT_NAME=${ORIG_COMPONENT_NAME%%:*}
+COMPONENT_VERSION=${ORIG_COMPONENT_NAME##*:}
+HASH_ALG="${ORIG_COMPONENT_VERSION%%:*}"
+case ${HASH_ALG^^} in
+  SHA256) COMPONENT_HASH_ALG="SHA-256"
+          ;;
+  *)      echo >&2 "Unknonwn hash algorithm $HASH_ALG"
+esac
+COMPONENT_HASH_CONTENT="${ORIG_COMPONENT_VERSION##*:}"
+
+echo "metadata:"
+echo "  tools:"
+echo "    tool:"
+echo "      vendor: $TOOL_VENDOR"
+echo "      name: $TOOL_NAME"
+echo "      version: $TOOL_VERSION"
+echo "      hashes:"
+echo "        hash:"
+echo "          alg: $TOOL_HASH_ALG"
+echo "          content: $TOOL_HASH_CONTENT"
+echo "  authors:"
+echo "    author:"
+echo "      name: $AUTHOR_NAME"
+echo "      email: $AUTHOR_EMAIL"
+echo "  component:"
+echo "    supplier:"
+echo "      name: $SUPPLIER_NAME"
+echo "      url: $SUPPLIER_URL"
+echo "    author: $COMPONENT_AUTHOR_NAME"
+echo "    name: $ORIG_COMPONENT_NAME -> $COMPONENT_NAME"
+echo "    version: $ORIG_COMPONENT_VERSION -> $COMPONENT_VERSION"
+echo "    hashes:"
+echo "      hash:"
+echo "        alg: $COMPONENT_HASH_ALG"
+echo "        content: $COMPONENT_HASH_CONTENT"
+
+[ -z "$TOOL_VENDOR" ] && echo >&2 "Unable to determine SBOM tool vendor" && exit 1
+[ -z "$TOOL_NAME" ] && echo >&2 "Unable to determine SBOM tool name" && exit 1
+[ -z "$TOOL_VERSION" ] && echo >&2 "Unable to determine SBOM tool version" && exit 1
+[ -z "$TOOL_HASH_ALG" ] && echo >&2 "Unable to determine SBOM tool hash algorithm" && exit 1
+[ -z "$TOOL_HASH_CONTENT" ] && echo >&2 "Unable to determine SBOM tool hash content" && exit 1
+[ -z "$AUTHOR_NAME" ] && echo >&2 "Unable to determine SBOM author name" && exit 1
+[ -z "$AUTHOR_EMAIL" ] && echo >&2 "Unable to determine SBOM author email" && exit 1
+[ -z "$SUPPLIER_NAME" ] && echo >&2 "Unable to determine component supplier name" && exit 1
+[ -z "$SUPPLIER_URL" ] && echo >&2 "Unable to determine component supplier url" && exit 1
+[ -z "$COMPONENT_AUTHOR_NAME" ] && echo >&2 "Unable to determine component author name" && exit 1
+[ -z "$COMPONENT_NAME" ] && echo >&2 "Unable to determine component name" && exit 1
+[ -z "$COMPONENT_VERSION" ] && echo >&2 "Unable to determine component version" && exit 1
+[ -z "$COMPONENT_HASH_ALG" ] && echo >&2 "Unable to determine component hash algorithm" && exit 1
+[ -z "$COMPONENT_HASH_CONTENT" ] && echo >&2 "Unable to determine component hash content" && exit 1
+
+PATCHED_OUTPUT="${OUTPUT}.patched"
+
+python3 <(cat <<END
```

Comment:
maybe add a check for python3 at top of file ?

Reply:
done, thanks